### ISSUE TITLE
Error  Uncaught TypeError: Converting circular structure to JSON

### ISSUE THREAD

**Issue by DelfsEngineering** [robinvdvleuten/vuex-persistedstate#41]
Using the basic example I am getting a circular structure error
I suspect the actual issue is elsewhere but I don't know where to look.

```// VUEX

import Vuex from 'vuex';
import createPersistedState from 'vuex-persistedstate'

// Added Modules
import auth from './modules/auth'; 
import site from './modules/site'; 

const store = new Vuex.Store({
  modules: {
    auth,
    site
  },
  state: {},
  plugins: [createPersistedState()]  
});

export default store;
```

**Comment by robinvdvleuten** [comment]
@DelfsEngineering please provide some more context of your JSON structure. Without any knowledge on how your modules look like, it's a very hard issue to solve.

**Comment by robinvdvleuten** [comment]
No update by @DelfsEngineering on the issue. Please see the jsfiddle links in the readme for a basic running example of the plugin.

**Comment by riksnelders** [comment]
![image](https://user-images.githubusercontent.com/15074808/51470254-ad73c200-1d73-11e9-982d-c03c92f10a08.png)
 crashes it too

**Comment by itsbibsbarreto** [comment]
I was having this error, I was importing the lib like that:

`plugins: [createPersistedState()],`

I changed to: 

`plugins: [createPersistedState],`

and it worked!!

**Comment by daweiyong** [comment]
It seems that it is caused by the matched attribute of vue-router. See how the author can improve it. The framework used is vue-element-admin. You can clone to see the error message;

**Comment by daweiyong** [comment]
> No update by @DelfsEngineering on the issue. Please see the jsfiddle links in the readme for a basic running example of the plugin.

It seems that it is caused by the matched attribute of vue-router. See how the author can improve it. The framework used is vue-element-admin. You can clone to see the error message;

**Comment by runryan** [comment]
我也遇到了这个问题。最开始我分析自己的 state 的时候时候，我确认它是不会出现 **circular structure to JSON** 这个问题的。

于是我打了断点调试，发现在提交 vuex store 的状态时，这个错误出现在了vuex-persistedstate 调用 JSON.stringfy(store.state)方法。

我猜测state 本身有问题，而不是vuex-persistedstate的原因。所以我尝试了直接去打印 JSON.stringfy(store.state)，遇到了同样的错误。

那么怎样验证我的 state 是否有错呢，我尝试了[这个方法](https://stackoverflow.com/questions/11616630/how-can-i-print-a-circular-structure-in-a-json-like-format)打印了 $store.state 的字符串数据，发现有很多不是我想要的数据，也就是我在处理业务逻辑时，没有考虑到位，塞进了许多可能导致 JSON.stringfy() 方法失效的数据。这也证实了我的猜测。

移除了冗余的数据后，这个问题顺利解决了。

In short, make sure vuex state itself can be stringfied be `JSON` correctly.
